### PR TITLE
feat: state management

### PR DESCRIPTION
**Pull request by anamontiaga**
Closes https://github.com/Vizzuality/front-end-scaffold/issues/100

It would be recommended to add plugin to generate debug labels for dev tools: [https://jotai.org/docs/api/swc](https://jotai.org/docs/api/swc)

### CODE REVIEW

**Comment by andresgnlez** [comment]
Looks like the [debug tools](https://jotai.org/docs/api/swc#swc-jotai-debug-label) do not get along with `next@12` but they do with `next@13`. Considering that, https://github.com/Vizzuality/front-end-scaffold/pull/106 might be a blocker until is merged.

File: src/components/forms/component.stories.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable prettier/prettier */
```
Comment (davidsingal) [inline src/components/forms/component.stories.tsx:1]: why disabling prettier?

File: src/components/jotai-provider/component.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Provider } from 'jotai';
+import { useAtomsDevtools } from 'jotai/devtools';
+
+const AtomsDevtools = ({ children }) => {
+  useAtomsDevtools('dev');
```
Comment (davidsingal) [inline src/components/jotai-provider/component.tsx:5]: could we set `dev`only for development env??

File: src/containers/filter-with-hash/component.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/* eslint-disable prettier/prettier */
```
Comment (davidsingal) [inline src/containers/filter-with-hash/component.stories.tsx:1]: same, why disabling prettier??

**Comment by davidsingal** [comment]
What is the status of this PR??

**Comment by anamontiaga** [comment]
> What is the status of this PR??

@davidsingal apart from dev tools tiny changes, we should agree if @barbara-chaves approach could be used.

**Comment by andresgnlez** [comment]
Looks like our winner is Recoil. We will need to update this somewhere but I think, for now, we can close this PR. Thank you both for testing this!

**Pull request closed without merging**